src/librustc_codegen_ssa/mir/debuginfo.rs

```diff
@@ -341,12 +357,28 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                 )
             });
 
-            per_local[var.place.local].push(PerLocalVarDebugInfo {
-                name: var.name,
-                source_info: var.source_info,
-                dbg_var,
-                projection: var.place.projection,
-            });
+            match var.value {
+                mir::VarDebugInfoContents::Place(place) => {
+                    per_local[place.local].push(PerLocalVarDebugInfo {
+                        name: var.name,
+                        source_info: var.source_info,
+                        dbg_var,
+                        projection: place.projection,
+                    });
+                }
+                mir::VarDebugInfoContents::Const(c) => {
+                    if let (Some(scope), Some(dbg_var)) = (scope, dbg_var) {
+                        let const_val =
+                            if let ConstKind::Value(value) = c.val { value } else { continue };
+                        let operand = OperandRef::from_const(bx, const_val, var_ty);
```

Comment:
Not sure if this is available here, but there's https://doc.rust-lang.org/nightly/nightly-rustc/rustc_codegen_ssa/mir/struct.FunctionCx.html#method.eval_mir_constant_to_operand which can to from a constant directly to an `Operand`. You may need to adjust things so it doesn't look at `mir::Constant` but at `ty::Const`